File: src/main/java/src/Calendario/Tarea.java
package Calendario;

import personas.*;

public class Tarea {
	
	//atributos
	protected String descipcion;
	protected Materia materia;
	protected String fecha_Entrega;
	protected String fecha_Inicio;
	protected double calificacion;
	private Estudiante estudiante;
	
	//constructor
	public Tarea(String descripcion, Materia materia, String fecha_Entrega, String fecha_Inicio, double calificacion,Estudiante estudiante) {
		this.descipcion =descripcion;
		this.materia = materia;
		this.fecha_Entrega = fecha_Entrega;
		this.fecha_Inicio = fecha_Inicio;
		this.calificacion = calificacion;
		this.estudiante = estudiante;
	}
	//constructor 2
	public Tarea(Materia materia, String fecha_Entrega, double calificacion) {
		fecha_Inicio = "fecha";
		//calificaciones
		this.materia = materia;
		this.fecha_Entrega = fecha_Entrega;
		this.calificacion = calificacion;
		ingresarTarea(this);
	}
	
	public void setDescripcion(String descripcion) {
		this.descipcion=descripcion;
	}
	
	public String getDescripcion() {
		return descipcion;
	}
	
	public void setMateria(Materia materia) {
		this.materia=materia;
	}
	
	public Materia getMateria() {
		return materia;
	}
	
	public void setFecha_Entrega(String fecha_Entrega) {
		this.fecha_Entrega=fecha_Entrega;
	}
	
	public String getFecha_Entrega() {
		return fecha_Entrega;
	}
	
	public void setFecha_Intrega(String fecha_Inicio) {
		this.fecha_Inicio=fecha_Inicio;
	}

	public String getFecha_Inicio() {
		return fecha_Inicio;
	}
	
	public void setCalificacion(double calificacion) {
		this.calificacion=calificacion;
	}
	
	public double getCalificacion() {
		return calificacion;
	}
	
	public static void ingresarTarea(Tarea tarea) {
		Materia.tareas_de_materia.add(tarea);
	}
	
	public static void eliminarTarea(Tarea tarea) {
		Materia.tareas_de_materia.remove(tarea);
	}

}

File: src_py/Calendario/Tarea.py
from Calendario.TareaEstudiante import TareaEstudiante
from typing import List

class Tarea:
    def __init__(self, descripcion: str="", fecha_entrega: str="", fecha_inicio: str=""):
        self.__descripcion = descripcion
        self.__fecha_entrega = fecha_entrega
        self.__fecha_inicio = fecha_inicio
        self.__tarea_estudiantes = []

    def getTarea_estudiantes(self):
        return self.__tarea_estudiantes

    def setTarea_estudiantes(self, tarea_estudiantes):
        self.__tarea_estudiantes = tarea_estudiantes

    def getDescripcion(self):
        return self.__descripcion

    def setDescripcion(self, descripcion):
        self.__descripcion = descripcion

    def getFecha_entrega(self):
        return self.__fecha_entrega

    def setFecha_entrega(self, fecha_entrega):
        self.__fecha_entrega = fecha_entrega

    def getFecha_inicio(self):
        return self.__fecha_inicio

    def setFecha_inicio(self, fecha_inicio):
        self.__fecha_inicio = fecha_inicio

    def set_grade(self, estudiante, grade):
        found = False
        for tarea_estudiante in self.__tarea_estudiantes:
            if tarea_estudiante.getEstudiante() == estudiante:
                tarea_estudiante.setGrade(grade)
                found = True
                break
        if not found:
            self.__tarea_estudiantes.append(TareaEstudiante(self, estudiante, grade))


    def get_grade(self, estudiante):
        for tarea_estudiante in self.__tarea_estudiantes:
            if tarea_estudiante.getEstudiante() == estudiante:
                return tarea_estudiante.getGrade()
        return 0.0

    def __str__(self):
        return f'Tarea sobre {self.descripcion}'
